LoadingUnloading.lua

```diff
@@ -0,0 +1,506 @@
+					if not spec:getCanDischargeToObject(currentDischargeNode) then
+						for i=1,#spec.dischargeNodes do
+							if spec:getCanDischargeToObject(spec.dischargeNodes[i])then
+								spec:setCurrentDischargeNodeIndex(spec.dischargeNodes[i]);
+								currentDischargeNode = spec:getCurrentDischargeNode()
+								break
+							end
+						end
+					end
+					if spec:getCanDischargeToObject(currentDischargeNode) and not rootVehicle.cp.driver:isNearFillPoint() then
+						if not object:getFillUnitFillType(currentDischargeNode.fillUnitIndex) or rootVehicle.cp.driver:ignoreTrigger() then 
+							return
+						end
+						spec:setDischargeState(Dischargeable.DISCHARGE_STATE_OBJECT)				
+						rootVehicle.cp.driver:setUnloadingState(object)
+					end
+				end
+			end
+		else
+			rootVehicle.cp.driver:countTriggerDown(object)
+		end
+	end
+	for _,impl in pairs(object:getAttachedImplements()) do
+		courseplay:isUnloadingTriggerAvailable(impl.object)
+	end
+end
+
+--loading/unloading text for mode 8
+function courseplay:dischargeToObject(superFunc,dischargeNode, emptyLiters, object, targetFillUnitIndex)
+	local dischargedLiters = superFunc(self,dischargeNode, emptyLiters, object, targetFillUnitIndex)
+	local rootVehicle = self:getRootVehicle()
+	if courseplay:checkAIDriver(rootVehicle) and dischargedLiters~=0 then
+		if not rootVehicle.cp.driver:is_a(FieldSupplyAIDriver) then 
+			return dischargedLiters
+		end
+		local fillType = self:getDischargeFillType(dischargeNode)
+		if object and object:isa(Vehicle) then
+			local objectRootVehicle = object:getRootVehicle()
+			if courseplay:checkAIDriver(objectRootVehicle) then
+			--	courseplay.debugFormat(2,"Mode 4 Driver found, dischargedLiters: "..dischargedLiters)
+				local fillLevel = object:getFillUnitFillLevel(targetFillUnitIndex)
+				local fillCapacity = object:getFillUnitCapacity(targetFillUnitIndex)
+				objectRootVehicle.cp.driver:setLoadingText(fillType,fillLevel,fillCapacity)
+			end
+		end
+		local fillLevel = self:getFillUnitFillLevel(dischargeNode.fillUnitIndex)
+		local fillCapacity = self:getFillUnitCapacity(dischargeNode.fillUnitIndex)
+		rootVehicle.cp.driver:setUnloadingText(fillType,fillLevel,fillCapacity)
+	end
+	return dischargedLiters
+end
+Dischargeable.dischargeToObject = Utils.overwrittenFunction(Dischargeable.dischargeToObject,courseplay.dischargeToObject)
+
+--LoadTrigger callback used to open correct cover for loading 
+function courseplay:loadTriggerCallback(triggerId, otherId, onEnter, onLeave, onStay, otherShapeId)
+	--legancy code!!!
+	courseplay:SiloTrigger_TriggerCallback(self, triggerId, otherActorId, onEnter, onLeave, onStay, otherShapeId)
+	
+	local fillableObject = g_currentMission:getNodeObject(otherId)
+	local rootVehicle
+	if fillableObject then 
+		rootVehicle = fillableObject:getRootVehicle()
+	end
+	if courseplay:checkAIDriver(rootVehicle) then
+		continue =true
+		if not rootVehicle.cp.driver:is_a(FillableFieldworkAIDriver) then
```

Comment:
Again, this would be prettier if there was an AIDriver:isTriggerCallbackEnabled() or something, returning false, and then a FillableFieldworkAIDriver:isTriggerCallbackEnabled(), returning true.